### PR TITLE
Add EUDEMO divertor builder

### LINKED ISSUE #841: Add EU-DEMO divertor builder
## Description of issue / requirement to address

We need to generate the geometry for an EU-DEMO like divertor space reservation

## Proposed solution

Add a builder for an EU-DEMO like divertor. Probably a couple of options to consider when it comes to make the divertor segmentation and 3-D CAD.

## Alternative solutions

N/A

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #841 

## Description

image

## Interface Changes

If you've had to update an interface or introduce a new interface as part of your change then let us know here.

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

**Comment by CoronelBuendia** [comment]
Example doesn't look great at the moment because of #901 and the fact that it's still pretty sensitive to inputs!

**Comment by CoronelBuendia** [comment]
> Example doesn't look great at the moment because of #901 and the fact that it's still pretty sensitive to inputs!

In fact the tests will fail until such a time that that is fixed... I want to get the procedure flowing first before making it robust to inputs.

File: bluemira/builders/EUDEMO/reactor.py [hunk 1]
```diff
@@ -260,7 +268,7 @@ def build_in_vessel_component_shapes(self, component_tree: Component):
         bluemira_print(f"Starting design stage: {name}")
 
         default_variables_map = {
-            "x1": {"value": "r_fw_ib_in"},  # ib radius
+            "x1": {"value": "r_fw_ib_in", "fixed": True},  # ib radius
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/reactor.py:271]: We want to pin this, and probably also the outer radius. There are some robustness issues at the moment

File: bluemira/builders/EUDEMO/ivc/ivc.py [hunk 2]
```diff
@@ -322,10 +350,8 @@ def build_ivc_xz_shapes(
     blanket_boundary = _extract_wire(
         components, BlanketThicknessBuilder.COMPONENT_BOUNDARY
     )
-    filled_blanket_face = BluemiraFace(blanket_boundary, label="blanket_face")
     plasma_facing_wire = _build_plasma_facing_wire(components)
-    plasma_facing_face = BluemiraFace(plasma_facing_wire, label="plasma_facing_face")
-    in_vessel_face = boolean_cut(filled_blanket_face, [plasma_facing_face])[0]
+    in_vessel_face = BluemiraFace([blanket_boundary, plasma_facing_wire])
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/ivc.py:354]: One of the nice things about primitives is we can make faces directly from two wires; no 2-D boolean cutting required

File: bluemira/builders/EUDEMO/ivc/divertor.py [hunk 3]
```diff
@@ -267,12 +278,15 @@ def _make_inner_baffle(
             inner_target_start = self._get_wire_end_with_largest(target, "x")
         return self.make_baffle(
             label=self.COMPONENT_INNER_BAFFLE,
-            start=np.array([x_lim, self.x_points[0].z]),
+            start=np.array([x_lim, z_lim]),
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/divertor.py:281]: Taking the x-point is not reliable for some reason. This amounts to the same thing and in fact gives us the flexibility to not build the divertor at the x-point z (which will be part of some future parameterisation)

File: bluemira/builders/EUDEMO/ivc/ivc.py [hunk 4]
```diff
@@ -282,9 +294,27 @@ def _make_flux_surface_keep_out_zone(self, psi_n: float) -> BluemiraWire:
         Make a "keep-out zone" from an equilibrium's flux surface.
         """
         flux_surface_zone = self.equilibrium.get_flux_surface(psi_n)
-        flux_surface_zone = make_polygon(flux_surface_zone.xyz, closed=True)
+        # Chop the flux surface to only take the upper half
+        indices = np.where(flux_surface_zone.z >= self.o_points[0][1])[0]
```
Comment (hsaunders1904) [inline bluemira/builders/EUDEMO/ivc/ivc.py:298]: Not sure that you need the `np.where` here, you can index directly with the boolean array

```suggestion
        indices = flux_surface_zone.z >= self.o_points[0][1]
```

**Review by hsaunders1904 — APPROVED** [review]
All looks good to me. Tiny note, but not an issue

**Pull request merged**